scripts/executors/src/start.ts

```diff
@@ -0,0 +1,294 @@
+      return 'web-components';
+    }
+
+    return '';
+  }
+}
+
+async function getSelectedTarget(
+  projects: Record<string, ProjectGraphProjectNode>,
+  selectedProject: string,
+  targetsDescription: Record<string, string>,
+  omitTargets: string[],
+  createDescription: TargetDescriptionFn,
+): Promise<string> {
+  const projectConfig = projects[selectedProject];
+  const projectTargets = projectConfig.data.targets ?? {};
+
+  const availableTargets = Object.keys(projectTargets)
+    .filter(targetName => {
+      return omitTargets.includes(targetName) ? false : true;
+    })
+    .concat('help');
+
+  const targetChoices = availableTargets.map(targetName => ({
+    name: formatOutput(
+      targetName,
+      createDescription(projectConfig, targetName, targetsDescription),
+      getLongestStringValue(availableTargets),
+    ),
+    value: targetName,
+  }));
+
+  const targetPrompt = new AutoComplete({
+    name: 'target',
+    message: 'Select target to run',
+    choices: targetChoices,
+    suggest,
+    limit: 5,
+    footer() {
+      return output.dim('(Scroll up and down to reveal more choices)');
+    },
+  });
+
+  return targetPrompt.run();
+}
+
+function createTargetDescription(
+  projectConfig: ProjectGraphProjectNode,
+  targetName: string,
+  targetsDescription: Record<string, string>,
+) {
+  const description = targetsDescription[targetName];
+  const nxTargetConfiguration = projectConfig.data.targets?.[targetName];
+
+  if (!nxTargetConfiguration) {
+    return description;
+  }
+
+  if (targetName === 'start') {
+    const scriptContent = nxTargetConfiguration.metadata?.scriptContent;
+    if (scriptContent.includes('storybook')) {
+      return `Start the project (Alias of "storybook" target)`;
+    }
+    return description;
+  }
+
+  if (targetName === 'e2e') {
+    const scriptContent = nxTargetConfiguration.metadata?.scriptContent;
+    const runnerType = getRunnerType(scriptContent);
+    return description + ` (using ${runnerType})`;
+  }
+
+  return description ?? nxTargetConfiguration.metadata?.scriptContent;
+
+  function getRunnerType(scriptContent: string): 'cypress' | 'playwright' {
+    if (scriptContent.includes('cypress')) {
+      return 'cypress';
+    }
+    if (scriptContent.includes('playwright')) {
+      return 'playwright';
+    }
+    throw new Error('invalid runner type');
+  }
+}
+
+function formatOutput(value: string, description: string, longestStringCharCount: number) {
+  const padding = ' '.repeat(longestStringCharCount - value.length);
+  return `${value}${padding} - ${description}`;
+}
+function getLongestStringValue(values: string[]) {
+  return values.reduce((acc, value) => {
+    return value.length > acc ? value.length : acc;
+  }, 0);
+}
+
+/**
+ *
+ * Filter autocomplete choices based on project name only.
+ */
+function suggest(typed: string, choices: Array<Choice>) {
+  const matches = choices.filter(choice => choice.value.includes(typed));
```

Comment:
filter only by project-names ( default is to use whole line - including our custom description. this is not ideal as when you'd write react-components it would not give you anything meaningful )